docs/Hooks.md

```diff
@@ -53,6 +54,35 @@ fastify.addHook('onRequest', async (request, reply) => {
 
 **Notice:** in the [onRequest](#onRequest) hook, `request.body` will always be `null`, because the body parsing happens before the [preValidation](#preValidation) hook.
 
+### preDecoding
+
+If you are using the `preDecoding` hook, you can transform the request payload stream before it is parsed. It receives the request and reply objects as other hooks, and a stream with the current request payload. It must return a stream.
+
+For instance, you can uncompress the request body:
+
+```js
+fastify.addHook('preDecoding', (request, reply, payload, done) => {
+  if(request.headers['content-encoding'] !== 'gzip') {
+    done(null, payload)
+  }
+
+  // Some code
+  done(null, payload.pipe(zlib.createGunzip()))
+})
+```
+Or `async/await`:
+```js
+fastify.addHook('preDecoding', async (request, reply, payload) => {
+  if(request.headers['content-encoding'] !== 'gzip') {
+    return payload
+  }
+
+  return payload.pipe(zlib.createGunzip())
+})
+```
+
+**Notice:** you should also add `receivedEncodedLength` property to the returned stream. This property is used to correctly match the request payload with the `Content-Length` header value. Ideally, this property should be updated on each received chunk. 
```

Comment:
I would put this in the example of async await since it is quite the same of the callback one